src/main/java/org/fundacionjala/coding/juan/EanValidator.java

```diff
@@ -0,0 +1,40 @@
+package org.fundacionjala.coding.juan;
+
+/**
+ * @author Juan Pablo
+ */
+public final class EanValidator {
+    /**
+     * CONSTRUCTOR.
+     */
+    private EanValidator() {
+
+    }
+    /**
+     * @param eanCode is the code provided as input.
+     *                @return  the validation.
+     */
+
+    public static boolean validating(final String eanCode) {
+        int sum = 0;
+        int checksum = 0;
+        final int finalNumber = 12;
+        final int multiply = 3;
+        final int finalFactor = 10;
+        final int aux = 48;
+        for (int i = 0; i < finalNumber; i++) {
+            if (i % 2 != 0) {
+                sum += ((eanCode.charAt(i) - aux) * multiply);
+            } else {
+                sum += (eanCode.charAt(i) - aux);
```

Comment:
you can have a variable for eanCode.charAt(i) - aux
because you are repeating in two lines 27 and 29.